File: loot_deck_test.py
########## Sabareesh Vishwanathan ##########
########## 112585006 #############
########## SAVISHWANATH #############
import time
import unittest
from threading import Thread

from loot import *

class TestLootDeck(unittest.TestCase):
    def test_create_merchant_ship(self):
         merchant_ship = MerchantShip(2)
         self.assertEqual(2, merchant_ship.value, "Value doesn't match")
         self.assertNotEqual(0, merchant_ship.value, "Value matches")

    def test_create_pirate_ship(self):
        pirateShip = PirateShip('green', 2)
        self.assertEqual(2, pirateShip.attack_value, "Value doesn't match")
        self.assertNotEqual(0, pirateShip.attack_value, "Value matches")
        self.assertEqual('green', pirateShip.color, "Color doesn't match")
        self.assertNotEqual('blue', pirateShip.color, "Color matches")

    def test_create_captain(self):
        captain = Captain('green')
        self.assertEqual('green', captain.color, "Color doesn't match")
        self.assertNotEqual('blue', captain.color, "Color matches")

    def test_create_deck_and_admiral(self):
        # deckie = Deck()
        deck = Deck().get_instance()
        for i in range(0, 25):
            self.assertTrue(isinstance(deck.cards[i], MerchantShip), 'Object is not of type MerchantShip')
        for i in range(25, 73):
            self.assertTrue(isinstance(deck.cards[i], PirateShip), 'Object is not of type PirateShip')
        for i in range(73, 77):
            self.assertTrue(isinstance(deck.cards[i], Captain), 'Object is not of type Captain')
        self.assertTrue(isinstance(deck.cards[77], Admiral), 'Object is not of type Admiral')
        with self.assertRaises(RuntimeError) as cm:
            Deck()
        self.assertEqual('Singleton', str(cm.exception))

    # def test_create_admiral(self):
    #     admiral = Admiral().get_instance()
    #     self.assertTrue(isinstance(admiral, Admiral), "Object is not of type Admiral")
    #     with self.assertRaises(RuntimeError) as cm:
    #         Admiral()
    #     self.assertEqual('Singleton', str(cm.exception))

File: loot_deck_test.rb
########## Sabareesh Vishwanathan ##########
########## 112585006 #############
########## SAVISHWANATH #############

require 'test/unit'
require 'test/unit/assertions'
require_relative "./loot.rb"

class MyTest < Test::Unit::TestCase
  def test_create_merchant_ship
    merchant_ship = MerchantShip.new(2)
    assert_equal(2, merchant_ship.get_value, "Value doesn't match")
    assert_not_equal(0, merchant_ship.get_value, "Value matches")
  end

  def test_create_pirate_ship
    pirateShip = PirateShip.new('green', 2)
    assert_equal(2, pirateShip.get_value, "Value doesn't match")
    assert_not_equal(0, pirateShip.get_value, "Value matches")
    assert_equal('green', pirateShip.get_color, "Color doesn't match")
    assert_not_equal('blue', pirateShip.get_color, "Color matches")
  end

  def test_create_captain
    captain = Captain.new('green')
    assert_equal('green', captain.get_color, "Color doesn't match")
    assert_not_equal('blue', captain.get_color, "Color matches")
  end
  def test_create_deck_and_admiral
    # deckie = Deck()
    deck = Deck.new
    for i in (0..24)
        assert_true(deck.cards[i].instance_of?(MerchantShip), 'Object is not of type MerchantShip')
    end
    for i in (25..72)
        assert_true(deck.cards[i].instance_of?(PirateShip), 'Object is not of type PirateShip')
    end
    for i in (73..76)
        assert_true(deck.cards[i].instance_of?(Captain), 'Object is not of type Captain')
    end
    assert_true(deck.cards[77].instance_of?(Admiral), 'Object is not of type Admiral')
  end
end
